### ISSUE TITLE
Addition of a build system generator

### ISSUE THREAD

**Issue by elfring** [aesophor/Wmderland#12]
I suggest to reuse a higher level build system than [your current make script](https://github.com/aesophor/Wmderland/blob/1d2c92d9631cad8e1dc973b77ac90b892d37923c/Makefile#L1 "Update candidate") so that powerful checks for software features will become easier.
- [CMake](https://cmake.org/ "Building software with CMake")
- [Autotools](https://www.gnu.org/software/autoconf/autoconf.html#introduction "Building software with Autoconf")

**Comment by aesophor** [comment]
I've added [CMakeLists.txt](https://github.com/aesophor/Wmderland/blob/staging/CMakeLists.txt).

I noticed that using the [old Makefile](https://github.com/aesophor/Wmderland/blob/staging/Makefile.old), the compiled binary is only 92 KB in size, while using CMake the binary is 668KB in size.

Using old Makefile (92 KB):
```
$ size Wmderland
   text	   data	    bss	    dec	    hex	filename
  55235	   2056	    384	  57675	   e14b	Wmderland
```

Using CMake (668 KB):
```
$ size Wmderland 
   text	   data	    bss	    dec	    hex	filename
 442400	   2096	    368	 444864	  6c9c0	Wmderland
```

So I think perhaps the .text section has included lots of unused symbols for debugging? But even if I set this in CMakeLists.txt, the binary size is still over 600 KB.
```
set(CMAKE_CXX_FLAGS "-std=c++14 -flto -Os -Wall -lpthread")
```

Any idea?

**Comment by aesophor** [comment]
Problem solved in 2880b55c68d4755b16eda07664d75a00e8b00428

First I moved the line `project(Wmderland)` to the first line of CMakeLists.txt, and then the compiled executable size drops from 668 KB to 143 KB miraculously :confused: 

Then I made the following changes:
* `cmake_minimum_required(VERSION 3.0)` -> `cmake_minimum_required(VERSION 3.9)`
* `set(CMAKE_BUILD_TYPE Release)` -> `set(CMAKE_BUILD_TYPE MinSizeRel)`

and the size drops to 92 KB!

Note: a CMake cheatsheet I've found http://www.brianlheim.com/2018/04/09/cmake-cheat-sheet.html
